pkg/telemetry/tracing.go

```diff
@@ -220,3 +231,40 @@ func newTextMapPropagator() propagation.TextMapPropagator {
 		propagation.Baggage{},
 	)
 }
+
+// IgnoredSpanProcessor processes spans and does not send them to the next
+// processor if they have the "ignored" attribute set to true.
+//
+// This can be useful for if a span's creation was required for propagation
+// purposes, but turned out not to be used, e.g. if idempotency filtered out a
+// queued item.
+type IgnoredSpanProcessor struct {
```

Comment:
I might want to build on top of this for span exporting, so would be nice to name it more generic.
Or we allow a list of span processors and just iterate through them.

Reply:
following your pattern here: https://github.com/inngest/inngest/pull/1248/files#diff-64030f762860cd37b119fd2737b96794faa0c08b5493a0223a1d740a1c01ada2R13-R45

Reply:
Ah, I created a `newSpanProcessor` function so that we can have one spot to build a stack of span processors common for each provider.

https://github.com/inngest/inngest/pull/1249/files/81f5620c59106b14d96d55384b14649d7b09f53d#diff-7eafc162c0f1d5cc4d869885af2f577b3bc7ea123a559f134e6102fda1cd366fR91-R95

Right now it's just:
- `IgnoredSpanProcessor`
- `BatchSpanProcessor`

But we can just add anything we need there.

Reply:
Discussed this in relation to #1248.